### ISSUE TITLE
Implement promise in a demo

### ISSUE THREAD

**Issue by raq929** [ga-wdi-boston/node-api-promises#12]
@gaand thinks there should be a demo of promise implementation. (ie, write our own promise constructor)
I am skeptical; I think this would be awesome for the top of the class and muddy the waters for people who are behind.

**Comment by payne-chris-r** [comment]
Yeah, I think it makes this talk 6 units instead of 3, but there is **definitely** pedagogical purpose to this...

**Comment by payne-chris-r** [comment]
I think #14 might be just as effective, but take much less time. People got hung up with the fact that `fs.writeFile` took a call back, and that callback was "the start" of our promise...

**Comment by jrhorn424** [comment]
I think there is pedagogical value here, and it should be a separate lesson. It may be better as a deep-dive, not an intro.

**Comment by gaand** [comment]
I think showing a simple implementation (as we do with ajax, sort of) after they've used promises might be clarifying for many if not most developers.

**Comment by raq929** [comment]
I changed my mind after explaining promises to Joel today. We should do this.

**Comment by raq929** [comment]
There is no time in cohort 016 to add another lesson. =/
